spec.html

```diff
@@ -32497,173 +32497,107 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
           1. Let _searchLength_ be the length of _searchString_.
           1. Let _position_ be ! StringIndexOf(_string_, _searchString_, 0).
           1. If _position_ is -1, return _string_.
-          1. Let _preserved_ be the substring of _string_ from 0 to _position_.
+          1. Let _preceding_ be the substring of _string_ from 0 to _position_.
+          1. Let _following_ be the substring of _string_ from _position_ + _searchLength_.
           1. If _functionalReplace_ is *true*, then
             1. Let _replacement_ be ? ToString(? Call(_replaceValue_, *undefined*, &laquo; _searchString_, 𝔽(_position_), _string_ &raquo;)).
           1. Else,
             1. Assert: Type(_replaceValue_) is String.
             1. Let _captures_ be a new empty List.
             1. Let _replacement_ be ! GetSubstitution(_searchString_, _string_, _position_, _captures_, *undefined*, _replaceValue_).
-          1. Return the string-concatenation of _preserved_, _replacement_, and the substring of _string_ from _position_ + _searchLength_.
+          1. Return the string-concatenation of _preceding_, _replacement_, and _following_.
         </emu-alg>
         <emu-note>
           <p>The `replace` function is intentionally generic; it does not require that its *this* value be a String object. Therefore, it can be transferred to other kinds of objects for use as a method.</p>
         </emu-note>
 
-        <emu-clause id="sec-getsubstitution" type="abstract operation">
+        <emu-clause id="sec-getsubstitution" type="abstract operation" oldids="table-replacement-text-symbol-substitutions,table-45">
           <h1>
             GetSubstitution (
               _matched_: unknown,
               _str_: unknown,
               _position_: a non-negative integer,
               _captures_: unknown,
               _namedCaptures_: unknown,
-              _replacement_: unknown,
+              _replacer_: unknown,
             )
           </h1>
           <dl class="header">
           </dl>
+          <p>For the purposes of this abstract operation, a <em>decimal digit</em> is a code unit in the range 0x0030 (DIGIT ZERO) to 0x0039 (DIGIT NINE) inclusive.</p>
           <emu-alg>
             1. Assert: Type(_matched_) is String.
             1. Let _matchLength_ be the number of code units in _matched_.
             1. Assert: Type(_str_) is String.
             1. Let _stringLength_ be the number of code units in _str_.
             1. Assert: _position_ &le; _stringLength_.
-            1. Assert: _captures_ is a possibly empty List of Strings.
-            1. Assert: Type(_replacement_) is String.
+            1. Assert: _captures_ is a possibly empty List, each of whose elements is a String or *undefined*.
+            1. Assert: Type(_replacer_) is String.
+            1. Let _preceding_ be the substring of _str_ from 0 to _position_.
```

Comment:
This seems like it could be inlined, especially if we take the suggestion for more concisely determining `_following_`.

Reply:
I did it this way because I liked the 'documentation' of the words "preceding" and "following", and also because they parallel the corresponding definitions in String.prototype.replace. But I can also see the argument of pushing the expression closer to its use. I'll go with editor consensus.

Reply:
I inlined `_preceding_` and `_following_`.

I also moved the definitions of `_matchLength_` and `_tailPos_` into the "following" case, since that's the only place they're used.